File: libsignal/service/src/main/java/org/whispersystems/signalservice/api/groupsv2/TemporalCredential.java
package org.whispersystems.signalservice.api.groupsv2;

import com.fasterxml.jackson.annotation.JsonProperty;

public class TemporalCredential {

  @JsonProperty
  private byte[] credential;

  @JsonProperty
  private int redemptionTime;

  public byte[] getCredential() {
    return credential;
  }

  public int getRedemptionTime() {
    return redemptionTime;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/groupsv2/TemporalCredential.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.groupsv2

class TemporalCredential(
  val credential: ByteArray,
  val redemptionTime: Long
)
